jekyll/_posts/2021-09-25-version-160-released.md

```diff
@@ -0,0 +1,839 @@
+```nim
+template fn(a: iterable) # or template fn[T](a: iterable[T])
+# instead of:
+template fn(a: untyped)
+```
+This solves a number of long standing issues related to iterators. In particular,
+iterable arguments can now be used with MCS, e.g. `iota(3).toSeq` now works.
+See PR [#17196](https://github.com/nim-lang/Nim/pull/17196) for additional details.
+
+
+## Private imports and private field access
+A new import syntax `import foo {.all.}` now allows to import all symbols (public or private) from `foo`.
+This can be useful for testing purposes or for more flexibility in project organization.
+
+Example:
+```nim
+from system {.all.} as system2 import nil
+echo system2.ThisIsSystem # ThisIsSystem is private in `system`
+import os {.all.} # weirdTarget is private in `os`
+echo weirdTarget # or `os.weirdTarget`
+```
+
+Added a new module `std/importutils`, and an API `privateAccess`, which allows access
+to private fields for an object type in the current scope.
+
+Example:
+```nim
+import times
+from std/importutils import privateAccess
+block:
+  let t = now()
+  # echo t.monthdayZero # Error: undeclared field: 'monthdayZero' for type times.DateTime
+  privateAccess(typeof(t)) # enables private access in this scope
+  echo t.monthdayZero # ok
+```
+
+See PR [#17706](https://github.com/nim-lang/Nim/pull/17706) for additional details.
+
+
+## `nim --eval:cmd`
+Added `nim --eval:cmd` to evaluate a command directly:, e.g.: `nim --eval:"echo 1"`.
+It defaults to `e` (nimscript) but can also work with other commands, e.g.:
+```bash
+find . | nim r --eval:'import strutils; for a in stdin.lines: echo a.toUpper'
+```
+You can now use nim as a calculator, e.g. `nim --eval:'echo 3.1 / (1.2+7)'`.
+You can also use it to explore a module's APIs, including private symbols:
+```bash
+nim --eval:'import os {.all.}; echo weirdTarget'
+```
+See PR [#15687](https://github.com/nim-lang/Nim/pull/15687) for more details.
+
+
+## Round-trip float to string
+`system.addFloat` and `system.$` now can produce string representations of floating point numbers that are minimal in size and possess round-trip and correct rounding guarantees (via the [Dragonbox](https://raw.githubusercontent.com/jk-jeon/dragonbox/master/other_files/Dragonbox.pdf) algorithm).
+This currently has to be enabled via `-d:nimPreviewFloatRoundtrip`.
+It is expected that this behavior becomes the new default in upcoming versions, as with other `nimPreviewX` define flags.
+
+Example:
+```nim
+from math import round
+let a = round(9.779999999999999, 2)
+assert a == 9.78
+echo a # with `-d:nimPreviewFloatRoundtrip`: 9.78, like in python3 (instead of  9.779999999999999)
+```
+
+## New `std/jsbigints` module
+Provides arbitrary precision integers for the JS target. See PR [#16409](https://github.com/nim-lang/Nim/pull/16409).
+Example:
+```nim
+import std/jsbigints
+assert 2'big ** 65'big == 36893488147419103232'big
+echo 0xdeadbeaf'big shl 4'big # 59774855920n
```

Comment:
```suggestion
echo 0xdeadbeef'big shl 4'big # 59774856944
```

Reply:
the `n` indicates bigints in js, see for yourself :)
```
nim r -b:js --eval:"import std/jsbigints; echo 0xdeadbeaf'big shl 4'big"
```

Reply:
Ah, it looked like a typo. The other change was `0xdeadbeaf` to `0xdeadbeef`, the latter is what people actually use.

Reply:
oh ya, missed that part, fixed